modules/dnn/test/test_model.cpp

```diff
@@ -202,6 +222,21 @@ TEST_P(Test_Model, DetectionMobilenetSSD)
                     scoreDiff, iouDiff, confThreshold, nmsThreshold, size, mean, scale);
 }
 
+TEST_P(Test_Model, Segmentation)
+{
+    std::string inp = _tf("dog416.png");
+    Net network = readNetFromONNX(_tf("segmentation.onnx"));
+    std::string exp = _tf("segmentation_exp.npy");
```

Comment:
Which model is used for test?

Reply:
Its just a dummy, small model that I created.

Reply:
So why not to test existing model? Check ENet from Torch test.

Reply:
@dkurt I get 404 error when downloading that model using the script in the opencv extra

Reply:
@dkurt  I changed the test data to use an fcn8 hope its ok now :)

Reply:
What is the source of `testdata/dnn/segmentation_exp.npy`? For this model there is already reference data.

Reply:
Is the expected output of the model for that image, when ran in caffe. I generated it

Reply:
It's 1MB! :) You can just test the same image from [TEST(Reproducibility_FCN, Accuracy)](https://github.com/opencv/opencv/blob/master/modules/dnn/test/test_caffe_importer.cpp).

Reply:
@dkurt The reference data in that example is just the output of the network, not the argmax over the channel dimension of the output. I am aware that I can just compute that after loading it, but what kind of test would that be. Testing the argmax code inside the module, against the same argmax code outside of it.

Reply:
For 500x500 CV_8UC1 mask? :confused: It must be even less using .png compression

Reply:
I can store it in a png file instead of an npy. Would that be OK?

Reply:
@dvd42, Even in NPY it should take about 250KB.

Reply:
@dkurt , done. Its 2.3kb now :)

Reply:
Just for experiment, can you try to run this model on lower input resolution? Will it produce maybe inaccurate but valid results?

Reply:
@dkurt, hey yeah I just tried it and the segmentation looks fine at half the resolution (256, 256)

Reply:
Thanks!